source/vibe/http/common.d

```diff
@@ -469,20 +488,25 @@ final class ChunkedOutputStream : OutputStream {
 	{
 		assert(!m_finalized);
 		if( m_buffer.data.length > 0 ) flush();
+		ubyte[] buffer = null;
+		auto getBytes = (size_t size) {
+			if (buffer is null || buffer.length < size)
+				buffer = new ubyte[](size);
+				return buffer[0..size];
+			};
```

Comment:
I think `AllocAppender` should definitely get extended. Either by adding an `append` method similar to the one you mentioned, or using the scheme that I mentioned in my reply to the wrong message thread:
> Flushing can indeed be removed here, it was just necessary to cleanly switch between buffered and stream modes. I just realized now that m_buffer is an AllocAppender. We'd need something like AllocAppender.put(scope size_t delegate(scope T[] dst) del);, where the delegate is called with m_remaining and the return value is the number of bytes written to that buffer.

The advantage of the callback based version would be that it could be made to work with `immutable(T)[]`, but using a method that just returns a buffer is of course much nicer to look at.